aiida/cmdline/utils/multi_line_input.py

```diff
@@ -31,12 +32,38 @@ def edit_pre_post(pre=None, post=None, summary=None):
     template = env.get_template('prepost.bash.tpl')
     summary = summary or {}
     summary = {k: v for k, v in summary.items() if v}
-    content = template.render(default_pre=pre or '', default_post=post or '', summary=summary)
+
+    # Define a separator that will be splitting pre- and post- execution
+    # parts of the submission script
+    separator = "#====================================================#\n" \
+                "#=               Post execution script              =#\n" \
+                "#=  I am acting a as separator, do not modify me!!! =#\n" \
+                "#====================================================#\n"
+
+    content = template.render(default_pre=pre or '', separator=separator, default_post=post or '', summary=summary)
     mlinput = click.edit(content, extension='.bash')
     if mlinput:
         import re
-        stripped_input = re.sub(r'(^#.*$\n)+', '#', mlinput, flags=re.M).strip('#')
-        pre, post = [text.strip() for text in stripped_input.split('#')]
+
+        # Splitting the text in pre- and post- halfs
+        try:
+            pre, post = mlinput.split(separator)
+        except ValueError as err:
+            if str(err) == "need more than 1 value to unpack":
+                raise InputValidationError("Looks like you modified the "
+                                           "separartor that should NOT be modified. Please be "
```

Comment:
small typo `separartor`. We should also catch this `InputValidationError` in all the `verdi` commands that call `ensure_scripts` e.g. line 194 of `cmd_computer`. Wrap those in try/catch and raise `click.BadParameter` like so:
```
        try:
            pre, post = ensure_scripts(kwargs.pop('prepend_text', ''), kwargs.pop('append_text', ''), kwargs)
        except InputValidationError as exception:
            raise click.BadParameter('invalid prepend and or append text: {}'.format(exception))
```

Reply:
done.